src/spring-cloud/azext_spring_cloud/_validators_enterprise.py

```diff
@@ -116,3 +121,48 @@ def validate_api_portal_instance_count(namespace):
     if namespace.api_portal_instance_count is not None:
         if namespace.api_portal_instance_count < 1:
             raise InvalidArgumentValueError("--api-portal-instance-count must be greater than 0")
+
+
+def validate_buildpack_binding_properties(namespace):
+    """ Extracts multiple space-separated properties in key[=value] format """
+    if isinstance(namespace.properties, list):
+        properties_dict = {}
+        for item in namespace.properties:
+            properties_dict.update(validate_tag(item))
+        namespace.properties = properties_dict
+
+
+def validate_buildpack_binding_secrets(namespace):
+    """ Extracts multiple space-separated secrets in key[=value] format """
+    if isinstance(namespace.secrets, list):
+        secrets_dict = {}
+        for item in namespace.secrets:
+            secrets_dict.update(validate_tag(item))
+        namespace.secrets = secrets_dict
+
+
+def validate_buildpack_binding_not_exist(cmd, namespace):
+    client = get_client(cmd)
+    try:
+        binding_resource = client.buildpack_binding.get(namespace.resource_group,
+                                                         namespace.service,
+                                                         DEFAULT_BUILD_SERVICE_NAME,
+                                                         namespace.builder_name,
+                                                         namespace.name)
+        if binding_resource is not None:
+            raise CLIError('buildpack Binding {} in builder {} already exists '
```

Comment:
Could you please use a specific error type instead of CLIError ?

Reply:
Sure.

Reply:
Updated.